src/SourceCode.Clay.Algorithms/Edge.cs

```diff
@@ -12,40 +12,51 @@
 
 namespace SourceCode.Clay.Algorithms
 {
-    [DebuggerDisplay("{From} -> {To}")]
+    [DebuggerDisplay("{ToString(),nq}")]
     public readonly struct Edge<T> : IEquatable<Edge<T>>, IStructuralEquatable
     {
+        private readonly IEqualityComparer<T> _equalityComparer;
+
         public T From { get; }
+
         public T To { get; }
 
-        public Edge(T from, T to)
+        public Edge(T from, T to, IEqualityComparer<T> equalityComparer)
         {
             From = from;
             To = to;
+            _equalityComparer = equalityComparer ?? EqualityComparer<T>.Default;
         }
 
-        public static bool operator ==(Edge<T> edge1, Edge<T> edge2) => edge1.Equals(edge2);
-
-        public static bool operator !=(Edge<T> edge1, Edge<T> edge2) => !(edge1 == edge2);
+        public Edge(T from, T to)
+            : this(from, to, null)
+        { }
 
         public override string ToString() => $"{From} -> {To}";
 
         public override bool Equals(object obj)
-            => obj is Edge<T> o && Equals(o);
+            => obj is Edge<T> other
+            && Equals(other);
 
         public bool Equals(Edge<T> other)
-            => EqualityComparer<T>.Default.Equals(From, other.From)
-            && EqualityComparer<T>.Default.Equals(To, other.To);
+            => _equalityComparer.Equals(From, other.From)
```

Comment:
Explicit comparer is more flexible than assuming .Default

Reply:
`var equalityComparer = _equalityComparer ?? EqualityComparer<T>.Default`.